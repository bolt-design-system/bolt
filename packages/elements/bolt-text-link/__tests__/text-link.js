import {
  isConnected,
  render,
  renderString,
  stopServer,
  html,
} from '../../../testing/testing-helpers';
import schema from '../text-link.schema';
const { display, valign } = schema.properties;

const timeout = 90000;

describe('textLink', () => {
  let page;

  afterAll(async () => {
    await stopServer();
    await page.close();
  });

  beforeEach(async () => {
    await page.evaluate(() => {
      document.body.innerHTML = '';
    });
  }, timeout);

  beforeAll(async () => {
    page = await global.__BROWSER__.newPage();
    await page.goto('http://127.0.0.1:4444/', {
      timeout: 0,
    });
  }, timeout);

  test('basic text link', async () => {
    const results = await render('@bolt-elements-text-link/text-link.twig', {
      content: 'This is a text link',
      attributes: {
        href: 'https://pega.com',
        target: '_blank',
      },
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });
});
