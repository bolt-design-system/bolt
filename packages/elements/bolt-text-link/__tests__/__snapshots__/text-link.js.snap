// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`textLink basic text link 1`] = `
<a href="https://pega.com"
   target="_blank"
   class="e-bolt-text-link"
>
  This is a text link
</a>
`;
